cipd_packages/codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -255,6 +255,10 @@ update these file paths accordingly.
       }
       log.info('Child file of directory ${directory.basename} is ${entity.basename}');
     }
+    final String directoryExtension = directory.basename.split('.').last;
+    if (directoryExtension == 'framework' || directoryExtension == 'xcframework') {
```

Comment:
Can this be implemented in a generic way? e.g. only check if the path is a directory rather than using the extension and hardcoding to framework or xcfrmamework

Reply:
Umm my understanding is that a generic directory might not qualify as a bundle.

Sorry for the verbosity, to summarize (3) in PR description, my understanding is that both .xcframework and .framework direcotry/folders qualify as bundles (based on info.plist and folder structure etc.), but a regular directory might not qualify (without info.plist and the required structure)

Reply:
He's right `.framework`s and `.xcframework`s are special with specific structures. Probably easier to check this faux "extension" than the bundle structure.